### PR TITLE
feat: create a reusable CI/CD workflow for Gradle

### PR DESCRIPTION
**Pull request by Judge40**
Create a reusable workflow to perform build and deploy for Gradle single module projects.

Create a template for the new workflow so it can be easily added to projects.

TIS21-3497
TIS21-3847

### CODE REVIEW

File: .github/workflows/build-gradle.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            echo ::set-output name=do-codeartifact-login::false
+          fi
+
+      - name: Configure AWS credentials
+        if: steps.check-codeartifact-secrets.outputs.do-codeartifact-login == 'true'
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.codeartifact-username }}
+          aws-secret-access-key: ${{ secrets.codeartifact-password }}
+          aws-region: eu-west-1
```
Comment (ReubenRobertsHEE) [inline .github/workflows/build-gradle.yml:64]: This region looks correct

Comment (Judge40) [reply .github/workflows/build-gradle.yml:64]: @ReubenRobertsHEE I'm not sure if you're confirming or questioning 😂 

Our current codeartifact repo only exists in eu-west-1, another case of it not being supported in eu-west-2 when we needed to use it.

File: .github/workflows/build-gradle.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+      - name: Add CodeArtifact env var
+        if: steps.check-codeartifact-secrets.outputs.do-codeartifact-login == 'true'
+        run: |
+          CODEARTIFACT_AUTH_TOKEN=`aws codeartifact get-authorization-token --domain hee --domain-owner 430723991443 --query authorizationToken --output text --duration-seconds 900`
+          echo CODEARTIFACT_AUTH_TOKEN=$CODEARTIFACT_AUTH_TOKEN >> $GITHUB_ENV
+          echo "::add-mask::$CODEARTIFACT_AUTH_TOKEN"
+
+      - name: Build
+        run: ./gradlew build
```
Comment (ReubenRobertsHEE) [inline .github/workflows/build-gradle.yml:74]: build = assemble + check ✔️

**Review by DorisWongNHS — APPROVED** [review]

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    name: Deploy to ${{ inputs.environment }}
+    environment: ${{ inputs.environment }}
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Create task definition
+        run: sed 's/${environment}/${{ inputs.environment }}/g' .aws/task-definition-template.json > ${{ runner.temp }}/task-definition.json
```
Comment (ReubenRobertsHEE) [inline .github/workflows/deploy.yml:33]: neat

File: .github/workflows/build-gradle.yml (hunk 1 above)
Comment (ReubenRobertsHEE) [reply .github/workflows/build-gradle.yml:64]: Yes, I was confirming - sorry, just working though the scripts and making sure I understand and have checked what I can :)

**Comment by Judge40** [comment]
> Not quite sure how to actually test - but I can't find anything obviously wonky 😁

Thanks, we'll find out when I try to use it 🙈

**Pull request merged**